gst/nnstreamer/tensor_filter/tensor_filter.c

```diff
@@ -457,6 +487,44 @@ gst_tensor_filter_transform (GstBaseTransform * trans,
       GST_STR_NULL (prop->model_files[0]));
   allocate_in_invoke = gst_tensor_filter_allocate_in_invoke (priv);
 
+  /* skip input data when throttling delay is set */
+  GST_OBJECT_LOCK (trans);
+
+  if (priv->throttling_delay != 0) {
+    GstClockTime curr_ts = GST_BUFFER_PTS (inbuf);
+    GstClockTime prev_ts = priv->prev_ts;
+
+    priv->prev_ts = curr_ts;
+
+    if (GST_CLOCK_TIME_IS_VALID (prev_ts)) {
+      GstClockTimeDiff diff = curr_ts - prev_ts;
+      GstClockTimeDiff delay;
+
+      priv->throttling_accum += diff;
+
+      /* check whether the average latency is longer than throttling delay */
+      delay = MAX (priv->prop.latency * 1000, priv->throttling_delay);
+
+      if (priv->throttling_accum < delay) {
```

Comment:
should this be `throttling_accum > delay`? throttling_accum indicates the time between consecutive frames. If this frame exceeds the allowed, then QoS is send. 
With the current code, if `curr_ts == prev_ts`, `throttling_accum` goes to 0, and a QoS event is sent upstream with positive diff as `delay` (from documentation - A positive diff value indicates how late the buffer with timestamp was.). However, in this case, the buffer is not at all late.

Reply:
> should this be throttling_accum > delay? throttling_accum indicates the time between consecutive frames. If this frame exceeds the allowed, then QoS is send.

No. `throttling_accum` is being used to drop frames. So, until it reaches to `delay`, all incoming frames should be dropped.

Reply:
Understood, I misunderstood buffer timestamp.